crates/router/src/connector/iatapay/transformers.rs

```diff
@@ -238,7 +241,7 @@ impl<F, T>
                         mandate_reference: None,
                         connector_metadata: None,
                         network_txn_id: None,
-                        connector_response_reference_id: None,
+                        connector_response_reference_id: connector_response_reference_id.clone(),
```

Comment:
```suggestion
                        connector_response_reference_id: item
                         .response
                         .merchant_payment_id
                         .or(item.response.iata_payment_id),
```
Checks are failing because this is not updated, can you please update this as well